container-search/src/main/java/com/yahoo/search/query/Select.java

```diff
@@ -0,0 +1,118 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.search.query;
+
+import com.yahoo.processing.request.CompoundName;
+import com.yahoo.search.Query;
+import com.yahoo.search.grouping.GroupingRequest;
+import com.yahoo.search.query.parser.ParserEnvironment;
+import com.yahoo.search.query.parser.ParserFactory;
+import com.yahoo.search.query.profile.types.FieldDescription;
+import com.yahoo.search.query.profile.types.QueryProfileType;
+import com.yahoo.search.yql.VespaGroupingStep;
+
+
+
+/**
+ * The parameters defining the where-clause and groping of a query
+ *
+ * @author henrhoi
+ */
+public class Select implements Cloneable {
```

Comment:
Ok, let's keep this, but with one change: All changes to the Select object should be reflected  If somebody calls query.getSelect().setWhere(...), that should actually take effect, i.e it should have the same effect as calling query.getModel().setQueryString.

Reply:
I will set the QueryTree to null and set the query's type to `SELECT`